workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/App.kt

```diff
@@ -31,14 +41,20 @@ import io.github.vinceglb.filekit.PlatformFile
 internal fun App(
   modifier: Modifier = Modifier
 ) {
+  var appWindowSize by remember { mutableStateOf(IntSize(0, 0)) }
   var selectedNode by remember { mutableStateOf<NodeUpdate?>(null) }
-  val workflowFrames = remember { mutableStateListOf<Node>() }
+  var frameSize by remember { mutableIntStateOf(0) }
+  var rawRenderPass by remember { mutableStateOf("") }
   var frameIndex by remember { mutableIntStateOf(0) }
   val sandboxState = remember { SandboxState() }
+  val nodeLocations = remember { mutableListOf<SnapshotStateMap<Node, Offset>>() }
```

Comment:
`nodeLocations` is read in composition, so it needs to be snapshot state backed. 
```suggestion
  val nodeLocations = remember { mutableStateListOf<SnapshotStateMap<Node, Offset>>() }
```

Reply:
Hm, the original goal was that I would only want compositions to track what's happening to each `SnapshotStateMap` since that's where all the locations are stored, and I would just index into the list. Would my current code not work for that?

Reply:
ah okay, that makes a lot of sense.